deploy/manifests/virtual-kubelet/base/deployment.yaml

```diff
@@ -103,10 +104,29 @@ spec:
           readOnly: true
         securityContext:
           privileged: true
+      - command:
```

Comment:
I do want to try attaching AWS and GCE cloud load balancers to services to make sure this doesn't break anything.  It _should_ work but I've said that about code a lot and been wrong a lot.

Reply:
> I do want to try attaching AWS and GCE cloud load balancers to services to make sure this doesn't break anything. It _should_ work but I've said that about code a lot and been wrong a lot.

That works on AWS:

    $ kubectl run nginx --image=nginx
    pod/nginx created
    $ kubectl expose pod nginx --port=80 --type=LoadBalancer
    service/nginx exposed
    $ kubectl get svc nginx
    NAME    TYPE           CLUSTER-IP       EXTERNAL-IP                                                               PORT(S)        AGE
    nginx   LoadBalancer   10.105.117.135   a64d230df08874c8399a98e115f0e3a2-1918165344.us-west-1.elb.amazonaws.com   80:32168/TCP   43m
    $ curl -fs a64d230df08874c8399a98e115f0e3a2-1918165344.us-west-1.elb.amazonaws.com | grep title
    <title>Welcome to nginx!</title>

However, the GCE cloud provider in k8s does not like the fact that there's no cloud instance called virtual-kubelet:

    default     3s          Normal    EnsuringLoadBalancer         service/nginx          Ensuring load balancer
    default     3s          Warning   CreatingLoadBalancerFailed   service/nginx          Error creating load balancer (will retry): failed to ensure load balancer for service default/nginx: instance not found

Whether we use host network mode or not does not matter though. It's hard-coded in the GCE cloud plugin to list all nodes and bail if any of them is not a GCE VM instance: https://github.com/kubernetes/legacy-cloud-providers/blob/master/gce/gce_loadbalancer_external.go#L59-L62

I'll create an issue for it.